sdks/python/apache_beam/runners/worker/worker_id_interceptor.py

```diff
@@ -0,0 +1,54 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Client Interceptor to inject worker_id"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import os
+import uuid
+
+import grpc
+
+
+class _ClientCallDetails(
+    collections.namedtuple('_ClientCallDetails',
+                           ('method', 'timeout', 'metadata', 'credentials')),
+    grpc.ClientCallDetails):
+  pass
+
+
+class WorkerIdInterceptor(grpc.StreamStreamClientInterceptor):
+
+  # Unique worker Id for this worker.
+  _worker_id = os.environ['WORKER_ID'] if os.environ.has_key(
```

Comment:
The ID I think you want is this one: https://github.com/apache/beam/blob/master/sdks/python/container/boot.go#L41

Reply:
Yes, For testing I am modifying the internal version of boot.go.
I was using woker_id to be more explicit but we can change it to just id.

Reply:
Sorry if I was unclear: this PR should also change the referenced boot.go code to set the WORKER_ID. The code here should also fail if that ID is not found -- the ID is set by the runner so that the gRPC servers can know who's who; they are not just unique IDs.

Reply:
Added the relevant boot.go changes

Reply:
Do we want to fail if WORKER_ID is not found?

Reply:
Do we really need to be backward compatible? This is mostly new code with no production usage. I would prefer to not have it succeed like this. But if you think this is necessary in the interim, we can add a TODO to remove the UUID generation.